File: core/src/main/java/pcb/uwu/units/composite/electromagnetism/ElectricChargeUnit.java
package pcb.uwu.units.composite.electromagnetism;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.quantity.ElectricCurrentUnit;
import pcb.uwu.units.quantity.TimeUnit;

public class ElectricChargeUnit extends CompositeUnit {

	public ElectricChargeUnit(ElectricCurrentUnit electricCurrentUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(electricCurrentUnit)
				.major(timeUnit));
	}
}

File: dimension/src/main/java/pcb/uwu/unit/derived/electromagnetism/ElectricChargeUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.ElectricCurrentUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.termodynamics.EnergyUnit

open class ElectricChargeUnit
    : CompositeUnit
{
    constructor(electricCurrentUnit: ElectricCurrentUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(electricCurrentUnit)
                        .major(timeUnit))

    constructor(energyUnit: EnergyUnit,
                electricPotentialUnit: ElectricPotentialUnit)
            : super(UnitCounter()
                        .major(energyUnit)
                        .major(electricPotentialUnit))
}